### PR TITLE
doc: add instructions for runing unit tests

### LINKED ISSUE #4574: doc: add instructions for how to run unit tests
## Enhancement
Add instructions for how to run unit tests in README
```
cd $BUILD
cmake $WORKSPACE/tiflash -GNinja -DCMAKE_BUILD_TYPE=DEBUG
ninja gtests_dbms
./dbms/gtests_dbms
```

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
### What problem does this PR solve?

Issue Number: close #4574 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -190,7 +190,14 @@ cmake $WORKSPACE/tiflash -DCMAKE_BUILD_TYPE=DEBUG
 
 ## Run Unit Tests
 
-TBD.
+To run unit tests, you need to build with `-DCMAKE_BUILD_TYPE=DEBUG`:
+
+```shell
+cd $BUILD
+cmake $WORKSPACE/tiflash -GNinja -DCMAKE_BUILD_TYPE=DEBUG
```
Comment (fuzhe1989) [inline README.md:197]:
```suggestion
cmake $WORKSPACE/tiflash -GNinja -DCMAKE_BUILD_TYPE=DEBUG -DENABLE_TESTS=ON
```

Comment (Lloyd-Pottiger) [reply README.md:197]: https://github.com/pingcap/tiflash/blob/master/CMakeLists.txt#L379
```cmake
# Enable tests by default when build type is debug
if (CMAKE_BUILD_TYPE_UC STREQUAL "DEBUG")
    set (ENABLE_TESTS_DEFAULT ON)
else ()
    set (ENABLE_TESTS_DEFAULT OFF)
endif ()
```

Comment (fuzhe1989) [reply README.md:197]: got it, thx

File: README.md [hunk 2]
```diff
@@ -190,7 +190,14 @@ cmake $WORKSPACE/tiflash -DCMAKE_BUILD_TYPE=DEBUG
...
+ninja gtests_dbms
+```
+And the executable unit test is in `$BUILD/dbms/gtests_dbms`.
```
Comment (fuzhe1989) [inline README.md:200]: better to mention `gtests_libcommon` too.

**Review by fuzhe1989 — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (Lloyd-Pottiger) [reply README.md:200]: ok, I will fix

**Comment by SchrodingerZhu** [comment]
@Lloyd-Pottiger this will skip some tests. Can you try also to show how to run all the tests (you can follow the CI process)?

**Comment by Lloyd-Pottiger** [comment]
> @Lloyd-Pottiger this will skip some tests. Can you try also to show how to run all the tests (you can follow the CI process)?

OK

File: README.md [hunk 3]
```diff
@@ -190,7 +190,16 @@ cmake $WORKSPACE/tiflash -DCMAKE_BUILD_TYPE=DEBUG
...
+To run unit tests, you need to build with `-DCMAKE_BUILD_TYPE=DEBUG`:
+
+```shell
+cd $BUILD
+cmake $WORKSPACE/tiflash -GNinja -DCMAKE_BUILD_TYPE=DEBUG
+ninja gtests_dbms
+ninja gtests_libcommon
+ninja gtests_libdaemon
+```
+And the executable unit test is in `$BUILD/dbms/gtests_dbms`, `$BUILD/libs/libcommon/src/tests/gtests_libcommon` and `$BUILD/libs/libdaemon/src/tests/gtests_libdaemon`.
```
Comment (SchrodingerZhu) [inline README.md:202]:
```suggestion
And the unit-test executables are at `$BUILD/dbms/gtests_dbms`, `$BUILD/libs/libcommon/src/tests/gtests_libcommon` and `$BUILD/libs/libdaemon/src/tests/gtests_libdaemon`.
```

**Review by SchrodingerZhu — APPROVED** [review]
LGTM with a minor suggestion.

**Pull request merged**